// @strapi/utils/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.traverse = exports.file = exports.importDefault = exports.convertQueryParams = exports.validateYupSchemaSync = exports.validateYupSchema = exports.errors = exports.forEachAsync = exports.reduceAsync = exports.mapAsync = exports.pipeAsync = exports.pagination = exports.providerFactory = exports.hooks = exports.setCreatorFields = exports.relations = exports.env = exports.contentTypes = exports.toKebabCase = exports.isCamelCase = exports.isKebabCase = exports.template = exports.stringEquals = exports.stringIncludes = exports.generateTimestampCode = exports.getAbsoluteServerUrl = exports.getAbsoluteAdminUrl = exports.keysDeep = exports.removeUndefined = exports.escapeQuery = exports.getConfigUrls = exports.getCommonBeginning = exports.nameToCollectionName = exports.joinBy = exports.startsWithANumber = exports.toRegressedEnumValue = exports.nameToSlug = exports.parseType = exports.traverseEntity = exports.sanitize = exports.parseMultipartData = exports.templateConfiguration = exports.policy = exports.handleYupError = exports.yup = void 0;
/**
 * Export shared utilities
 */
const parse_multipart_1 = __importDefault(require("./parse-multipart"));
exports.parseMultipartData = parse_multipart_1.default;
const parse_type_1 = __importDefault(require("./parse-type"));
exports.parseType = parse_type_1.default;
const policy = __importStar(require("./policy"));
exports.policy = policy;
const template_configuration_1 = __importDefault(require("./template-configuration"));
exports.templateConfiguration = template_configuration_1.default;
const validators_1 = require("./validators");
Object.defineProperty(exports, "yup", { enumerable: true, get: function () { return validators_1.yup; } });
Object.defineProperty(exports, "handleYupError", { enumerable: true, get: function () { return validators_1.handleYupError; } });
Object.defineProperty(exports, "validateYupSchema", { enumerable: true, get: function () { return validators_1.validateYupSchema; } });
Object.defineProperty(exports, "validateYupSchemaSync", { enumerable: true, get: function () { return validators_1.validateYupSchemaSync; } });
const errors = __importStar(require("./errors"));
exports.errors = errors;
const string_formatting_1 = require("./string-formatting");
Object.defineProperty(exports, "nameToSlug", { enumerable: true, get: function () { return string_formatting_1.nameToSlug; } });
Object.defineProperty(exports, "nameToCollectionName", { enumerable: true, get: function () { return string_formatting_1.nameToCollectionName; } });
Object.defineProperty(exports, "getCommonBeginning", { enumerable: true, get: function () { return string_formatting_1.getCommonBeginning; } });
Object.defineProperty(exports, "escapeQuery", { enumerable: true, get: function () { return string_formatting_1.escapeQuery; } });
Object.defineProperty(exports, "stringIncludes", { enumerable: true, get: function () { return string_formatting_1.stringIncludes; } });
Object.defineProperty(exports, "stringEquals", { enumerable: true, get: function () { return string_formatting_1.stringEquals; } });
Object.defineProperty(exports, "isKebabCase", { enumerable: true, get: function () { return string_formatting_1.isKebabCase; } });
Object.defineProperty(exports, "isCamelCase", { enumerable: true, get: function () { return string_formatting_1.isCamelCase; } });
Object.defineProperty(exports, "toRegressedEnumValue", { enumerable: true, get: function () { return string_formatting_1.toRegressedEnumValue; } });
Object.defineProperty(exports, "startsWithANumber", { enumerable: true, get: function () { return string_formatting_1.startsWithANumber; } });
Object.defineProperty(exports, "joinBy", { enumerable: true, get: function () { return string_formatting_1.joinBy; } });
Object.defineProperty(exports, "toKebabCase", { enumerable: true, get: function () { return string_formatting_1.toKebabCase; } });
const object_formatting_1 = require("./object-formatting");
Object.defineProperty(exports, "removeUndefined", { enumerable: true, get: function () { return object_formatting_1.removeUndefined; } });
Object.defineProperty(exports, "keysDeep", { enumerable: true, get: function () { return object_formatting_1.keysDeep; } });
const config_1 = require("./config");
Object.defineProperty(exports, "getConfigUrls", { enumerable: true, get: function () { return config_1.getConfigUrls; } });
Object.defineProperty(exports, "getAbsoluteAdminUrl", { enumerable: true, get: function () { return config_1.getAbsoluteAdminUrl; } });
Object.defineProperty(exports, "getAbsoluteServerUrl", { enumerable: true, get: function () { return config_1.getAbsoluteServerUrl; } });
const code_generator_1 = require("./code-generator");
Object.defineProperty(exports, "generateTimestampCode", { enumerable: true, get: function () { return code_generator_1.generateTimestampCode; } });
const contentTypes = __importStar(require("./content-types"));
exports.contentTypes = contentTypes;
const env_helper_1 = __importDefault(require("./env-helper"));
exports.env = env_helper_1.default;
const relations = __importStar(require("./relations"));
exports.relations = relations;
const set_creator_fields_1 = __importDefault(require("./set-creator-fields"));
exports.setCreatorFields = set_creator_fields_1.default;
const hooks = __importStar(require("./hooks"));
exports.hooks = hooks;
const provider_factory_1 = __importDefault(require("./provider-factory"));
exports.providerFactory = provider_factory_1.default;
const pagination = __importStar(require("./pagination"));
exports.pagination = pagination;
const sanitize_1 = __importDefault(require("./sanitize"));
exports.sanitize = sanitize_1.default;
const traverse_entity_1 = __importDefault(require("./traverse-entity"));
exports.traverseEntity = traverse_entity_1.default;
const async_1 = require("./async");
Object.defineProperty(exports, "pipeAsync", { enumerable: true, get: function () { return async_1.pipeAsync; } });
Object.defineProperty(exports, "mapAsync", { enumerable: true, get: function () { return async_1.mapAsync; } });
Object.defineProperty(exports, "reduceAsync", { enumerable: true, get: function () { return async_1.reduceAsync; } });
Object.defineProperty(exports, "forEachAsync", { enumerable: true, get: function () { return async_1.forEachAsync; } });
const convert_query_params_1 = __importDefault(require("./convert-query-params"));
exports.convertQueryParams = convert_query_params_1.default;
const import_default_1 = __importDefault(require("./import-default"));
exports.importDefault = import_default_1.default;
const template = __importStar(require("./template"));
exports.template = template;
const file = __importStar(require("./file"));
exports.file = file;
const traverse = __importStar(require("./traverse"));
exports.traverse = traverse;
const utils = {
    yup: validators_1.yup,
    handleYupError: validators_1.handleYupError,
    policy,
    templateConfiguration: template_configuration_1.default,
    parseMultipartData: parse_multipart_1.default,
    sanitize: sanitize_1.default,
    traverseEntity: traverse_entity_1.default,
    parseType: parse_type_1.default,
    nameToSlug: string_formatting_1.nameToSlug,
    toRegressedEnumValue: string_formatting_1.toRegressedEnumValue,
    startsWithANumber: string_formatting_1.startsWithANumber,
    joinBy: string_formatting_1.joinBy,
    nameToCollectionName: string_formatting_1.nameToCollectionName,
    getCommonBeginning: string_formatting_1.getCommonBeginning,
    getConfigUrls: config_1.getConfigUrls,
    escapeQuery: string_formatting_1.escapeQuery,
    removeUndefined: object_formatting_1.removeUndefined,
    keysDeep: object_formatting_1.keysDeep,
    getAbsoluteAdminUrl: config_1.getAbsoluteAdminUrl,
    getAbsoluteServerUrl: config_1.getAbsoluteServerUrl,
    generateTimestampCode: code_generator_1.generateTimestampCode,
    stringIncludes: string_formatting_1.stringIncludes,
    stringEquals: string_formatting_1.stringEquals,
    template,
    isKebabCase: string_formatting_1.isKebabCase,
    isCamelCase: string_formatting_1.isCamelCase,
    toKebabCase: string_formatting_1.toKebabCase,
    contentTypes,
    env: env_helper_1.default,
    relations,
    setCreatorFields: set_creator_fields_1.default,
    hooks,
    providerFactory: provider_factory_1.default,
    pagination,
    pipeAsync: async_1.pipeAsync,
    mapAsync: async_1.mapAsync,
    reduceAsync: async_1.reduceAsync,
    forEachAsync: async_1.forEachAsync,
    errors,
    validateYupSchema: validators_1.validateYupSchema,
    validateYupSchemaSync: validators_1.validateYupSchemaSync,
    convertQueryParams: convert_query_params_1.default,
    importDefault: import_default_1.default,
    file,
    traverse,
};
exports.default = utils;
